contracts/extensions/token/erc20/UnitToken.sol

```diff
@@ -0,0 +1,237 @@
+     */
+    function totalSupply() public view override returns (uint256) {
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        return bank.balanceOf(TOTAL, UNITS);
+    }
+
+    /**
+     * @dev Returns the amount of tokens owned by `account`.
+     */
+    function balanceOf(address account) public view override returns (uint256) {
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        return bank.balanceOf(account, UNITS);
+    }
+
+    /**
+     * @dev Returns the remaining number of tokens that `spender` will be
+     * allowed to spend on behalf of `owner` through {transferFrom}. This is
+     * zero by default.
+     *
+     * This value changes when {approve} or {transferFrom} are called.
+     */
+    function allowance(address owner, address spender)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        return _allowances[owner][spender];
+    }
+
+    /**
+     * @dev Sets `amount` as the allowance of `spender` over the caller's tokens.
+     *
+     * Returns a boolean value indicating whether the operation succeeded.
+     *
+     * IMPORTANT: Beware that changing an allowance with this method brings the risk
+     * that someone may use both the old and the new allowance by unfortunate
+     * transaction ordering. One possible solution to mitigate this race
+     * condition is to first reduce the spender's allowance to 0 and set the
+     * desired value afterwards:
+     * https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
+     *
+     * Emits an {Approval} event.
+     */
+    function approve(address spender, uint256 amount)
+        public
+        override
+        returns (bool)
+    {
+        address senderAddr = dao.getAddressIfDelegated(msg.sender);
+        require(
+            senderAddr != address(0),
+            "ERC20: approve from the zero address"
+        );
+        require(spender != address(0), "ERC20: approve to the zero address");
+        require(dao.isMember(senderAddr), "sender is not a member");
+
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        require(
+            bank.balanceOf(senderAddr, UNITS) > amount && amount > 0,
+            "sender does not have UNITS to transfer"
+        );
+
+        _allowances[senderAddr][spender] = amount;
+        emit Approval(senderAddr, spender, amount);
+        return true;
+    }
+
+    /**
+     * @dev Moves `amount` tokens from the caller's account to `recipient`.
+     *
+     * Returns a boolean value indicating whether the operation succeeded.
+     *
+     * Emits a {Transfer} event.
+     */
+     // Is called by the adapter: always
+    function transfer(address recipient, uint256 amount)
+        public
+        override
+        returns (bool)
+    {
+        address senderAddr = dao.getAddressIfDelegated(msg.sender);
+        require(dao.isMember(senderAddr), "sender is not a member");
```

Comment:
same as before. It should be configurable to be either open, memberOnly or closed for transfer. This is actually where we should do the checks

Reply:
re: If we do this in config -can we still do the following?   The ideal is to have it initially memberOnly transfer, and then if the DAO (either by admin or by vote) wants to open up transfers outside the DAO, then we could unpause the memberOnly.  One thought is to have a member only transfer that is always available and pausable/unpauseable external transfer.  wdyt?

Reply:
We added the config. By default the extension enables the member to member transfers. If you set the dao config: `unitTransfersType` to 1, it will enabled external transfers only. If you set that to 2, it will pause/block any transfers. And 0, enables member to member transfers again.

Reply:
Nevermind - Felipe introduced me to Configuration.sol